### PR TITLE
[WIP] vscode-extensions: auto-update many

### PR DESCRIPTION
**Pull request by kamadorueda**
###### Motivation for this change

Keeping extensions updated is normally a thing you want to do,
but I'm too lazy as to do it manually on a daily basis so I'm:
- Extending a little bit the current script with a feature flag that updates the code in-place and even do the commit
- Updating my extension set to their latest version
- If you agree, I could work on applying this strategy to more packages, similar to what `niv` and `nixpkgs-updated` do. With small refactors auto-updating is easy
- This PR seems big, but read it 1 commit at a time and it'll be easier
- 
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127741` at 814b40da run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built successfully:</summary>
  <ul>
    <li>vscode-extensions._4ops.terraform</li>
    <li>vscode-extensions.coenraads.bracket-pair-colorizer-2</li>
    <li>vscode-extensions.haskell.haskell</li>
    <li>vscode-extensions.ms-python.python</li>
    <li>vscode-extensions.ms-python.vscode-pylance</li>
    <li>vscode-extensions.ms-toolsai.jupyter</li>
    <li>vscode-extensions.tamasfe.even-better-toml</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@jonringer can you review this?

**Comment by jonringer** [comment]
do you mind adding a `doc/languages-frameworks/vscode.section.md` with some instructions on how to update the extensions?

**Comment by kamadorueda** [comment]
> do you mind adding a `doc/languages-frameworks/vscode.section.md` with some instructions on how to update the extensions?

Actually I really like the idea, I'm gonna work on it

**Comment by cideM** [comment]
@kamadorueda you can maybe take some inspiration from https://github.com/NixOS/nixpkgs/pull/113280#issuecomment-871937609 where I also wrote documentation on VSCode extensions.

For what it's worth, I'll maybe see if I can just add another shell script or so which let's users update extensions from STDIN rather than getting them from VSCode, which is a bit brittle. I think the current script doesn't even consider all possible binary names. 

So in other words, my typical use case is `"foo.bar" | update_ext` rather than always updating all of them at the same time.
